CoursesManager.UI/DataAccess/StudentDataAccess.cs

```diff
@@ -112,18 +112,62 @@ public void Add(Student student)
             }
         }
 
-        public void Delete(Student student)
+        public void Update(Student student)
         {
-            ArgumentNullException.ThrowIfNull(student);
+            try
+            {
+                // Update the address first
+                _addressDataAccess.Update(student.Address);
+
+                // Get the newly created address ID
+                int addressId = _addressDataAccess.GetLastInsertedId();
```

Comment:
Dit moet eigenlijk anders maar voor nu wel prima. Stel namelijk dat meerdere clients adressen inserten op hetzelfde moment is de last inserted id niet per definitie die id die jij hebt geinsert.

Reply:
Aangepast naar als het goed is is dit een betere verwerking ik had shadi's stukje overgenomen maar vergeten dat je bij update geen nieuwe addressid aanmaakt maar de huidige update int? addressId = student.AddressId; new MySqlParameter("@p_address_id", addressId)